src/Fixer/StringNotation/HeredocClosingMarkerFixer.php

```diff
@@ -0,0 +1,172 @@
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolver;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolverInterface;
+use PhpCsFixer\FixerConfiguration\FixerOptionBuilder;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\Preg;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+final class HeredocClosingMarkerFixer extends AbstractFixer implements ConfigurableFixerInterface
+{
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'Unify `heredoc` or `nowdoc` closing marker.',
+            [
+                new CodeSample(
+                    <<<'EOD'
+                        <?php $a = <<<"TEST"
+                        Foo
+                        TEST;
+
+                        EOD
+                ),
+                new CodeSample(
+                    <<<'EOD'
+                        <?php $a = <<<"TEST"
+                        Foo
+                        TEST;
+
+                        EOD,
+                    ['closing_marker' => 'EOF']
+                ),
+                new CodeSample(
+                    <<<'EOD'
+                        <?php $a = <<<"TEST"
+                        Foo
+                        TEST;
+
+                        EOD,
+                    ['explicit_heredoc_style' => false]
+                ),
+            ]
+        );
+    }
+
+    public function isCandidate(Tokens $tokens): bool
+    {
+        return $tokens->isTokenKindFound(T_START_HEREDOC);
+    }
+
+    protected function createConfigurationDefinition(): FixerConfigurationResolverInterface
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder(
+                'closing_marker',
+                'Preferred closing marker.'
+            ))
+                ->setAllowedTypes(['string'])
+                ->setDefault('EOD')
+                ->getOption(),
+            (new FixerOptionBuilder(
+                'reserved_closing_markers',
+                'Reserved closing markers to be kept unchanged.'
+            ))
+                ->setAllowedTypes(['array'])
+                ->setDefault([
+                    'CSS',
+                    'DIFF',
+                    'HTML',
+                    'JS',
+                    'JSON',
+                    'MD',
+                    'PHP',
+                    'PYTHON',
+                    'RST',
+                    'TS',
+                    'SQL',
+                    'XML',
+                    'YAML',
+                ])
+                ->getOption(),
+            (new FixerOptionBuilder(
+                'explicit_heredoc_style',
+                'Whether the closing marker should be wrapped in double quotes.'
+            ))
+                ->setAllowedTypes(['bool'])
+                ->setDefault(true)
```

Comment:
I didn't even know the `<<<"FOO"` syntax was possible.

`<<<FOO` seems to be the most widely used syntax. PER Coding Styles does not give requirements about this but examples only use the implicit syntax. Shouldn't we set the default value to `false`?

Reply:
Personally I would go with implicit too (no `"`), even though I prefer explicit format. We can always add this fixer to `@PhpCsFixer` with proper config, but for wider audience I believe `<<<EOD` is more fitting, as official docs mentions `"` quotings for heredoc somewhere at the end, as optional.

Reply:
👍, implemented.